src/psyclone/dynamo0p3.py

```diff
@@ -3338,13 +3344,12 @@ def _create_declarations(self, parent):
         :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
 
         '''
-        api_config = Config.get().api_conf("dynamo0.3")
         const = LFRicConstants()
         # Real scalar arguments
-        dtype = const.MAPPING_DATA_TYPES["gh_real"]
-        dkind = api_config.default_kind[dtype]
         for intent in FORTRAN_INTENT_NAMES:
             if self._real_scalars[intent]:
+                dtype = self._real_scalars[intent][0].intrinsic_type
```

Comment:
This is not my code so I am not sure what was the motivation. I think the main reason for grouping the scalars per intent was to save space by not having to generate an entire line for each declaration.
I think changing this is a separate issue as it involves modifying the scalar code generation in various tests, which I am happy to open and add a TODO here if you think it would be better.

Reply:
Let's leave it as it is for the moment. I don't think it is high enough priority to add another issue.